unit-tests/live/calib/test-advanced-tare-calibrations.py

```diff
@@ -0,0 +1,291 @@
+    if tare_json is None:
+        log.i('Using default parameters for Tare calibration.')
+        tare_json = '{\n  '+\
+                    '"host assistance": ' + str(int(host_assistance)) + ',\n'+\
+                    '"resize factor": 1\n'+\
+                    '}'
+    return tare_json
+
+
+def calculate_target_z():
+    number_of_images = 50  # The required number of frames is 10+
+    timeout_s = 30
+    target_size = [175, 100]
+
+    cfg = rs.config()
+    cfg.enable_stream(rs.stream.infrared, 1, 1280, 720, rs.format.y8, 30)
+
+    q = rs.frame_queue(capacity=number_of_images, keep_frames=True)
+    q2 = rs.frame_queue(capacity=number_of_images, keep_frames=True)
+    q3 = rs.frame_queue(capacity=number_of_images, keep_frames=True)
+
+    counter = 0
+
+    def cb(frame):
+        nonlocal counter
+        if counter > number_of_images:
+            return
+        for f in frame.as_frameset():
+            q.enqueue(f)
+            counter += 1
+
+    ctx = rs.context()
+    pipe = rs.pipeline(ctx)
+    pp = pipe.start(cfg, cb)
+    dev = pp.get_device()
+
+    try:
+        stime = time.time()
+        while counter < number_of_images:
+            time.sleep(0.5)
+            if timeout_s < (time.time() - stime):
+                raise RuntimeError(f"Failed to capture {number_of_images} frames in {timeout_s} seconds, got only {counter} frames")
+
+        adev = dev.as_auto_calibrated_device()
+        log.i('Calculating distance to target...')
+        log.i(f'\tTarget Size:\t{target_size}')
+        target_z = adev.calculate_target_z(q, q2, q3, target_size[0], target_size[1])
+        log.i(f'Calculated distance to target is {target_z}')
+    finally:
+        pipe.stop()
+
+    return target_z
+
+
+# Constants for validation
+HEALTH_FACTOR_THRESHOLD = 0.25
+TARGET_Z_MIN = 600
+TARGET_Z_MAX = 2000
+_target_z = None
+
+# Additional constants & thresholds for advanced calibration modification test
+PIXEL_CORRECTION = -1.5  # pixel shift to apply to principal point (right IR)
+EPSILON = 0.001          # distance comparison tolerance for reversion checks
+HEALTH_FACTOR_THRESHOLD_AFTER_MODIFICATION = 0.75  # OCC health factor acceptance for modified run
+
+
+def run_advanced_tare_calibration_test(host_assistance, image_width, image_height, fps, target_z=None):
+    """Run advanced tare calibration test with calibration table modifications.
+
+        Flow:
+            1. OCC (baseline) or Restore factory calibration
+            2. Read & log base principal point
+            3. Apply controlled raw intrinsic perturbation (ppy or ppx)
+            4. Verify perturbation was applied
+            5. Run OCC once and capture returned table
+            6. Write table, read final principal point, compute correction metrics
+    """
+    config, pipeline, calib_dev = get_calibration_device(image_width, image_height, fps)
+    try:
+
+        # 1. run tare for the first time to ensure we start from a known state
```

Comment:
We should leave each test with a device working properly. I think we can remove this initial calibration to reduce run time. Just make sure to restore calibration or reset to factory at end of each test.